### PR TITLE
Refactor for pipeline, add resiliency metric

### PR DESCRIPTION
**Pull request by ndrwhoang**

### CODE REVIEW

File: roc_bike_growth/metrics.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+from collections import deque
+
+# Resiliency metric
+# Taken from https://codereview.stackexchange.com/questions/184392/computing-resilience-of-the-network-presented-as-an-undirected-graph-in-python
+def resilience(graph, attack_order):
```
Comment (BakerAugust) [inline roc_bike_growth/metrics.py:5]: How is `attack_order` determined? Would a default k and random selection of nodes make sense?

File: notebooks/apply_gt.ipynb [hunk 2]
```diff
@@ -0,0 +1,86 @@
+{
```
Comment (BakerAugust) [inline notebooks/apply_gt.ipynb:1]: Is this the beginning of the analysis pipeline? What is this for?

File: roc_bike_growth/metrics.py (hunk 1 above)
Comment (BakerAugust) [reply roc_bike_growth/metrics.py:5]: Also, does it make sense to add an example into one of the notebooks (or add a new notebook?). I know we're not using formal tests, but that may help at least by showing expected behavior and a clear use case that can be maintained.

**Review by BakerAugust — COMMENTED** [review]
Thanks for adding these @ndrwhoang! Unsure if this is still in progress or ready to merge, but I added a few comments.

**Comment by BakerAugust** [comment]
@ndrwhoang this looks great and I'd love to get it merged-in if you think it's ready.

Are there meaningful changes in these notebooks?
image

Or is the new work just in `metrics.py` and `metric_testing.ipynb`?

**Comment by ndrwhoang** [comment]
Yeah the main stuff is in there, other changes are mostly from playing around with it.

**Pull request merged**